app/utils/orgs.js

```diff
@@ -68,4 +73,329 @@ const decryptOrgData = (orgKey, data) => {
   return tokenCrypt.decrypt(data, orgKey);
 };
 
-module.exports = { getOrg, verifyAdminOrgKey, encryptOrgData, decryptOrgData};
+
+const encryptStrUsingOrgEncKey = ({ str, org })=>{
+  if((org.encKeys||[]).length < 1){
+    return { data: str }; // lazy feature flag for now
+  }
+  // finds the first non-deleted key in org.encKeys
+  var encKey = _.find(org.encKeys||[], (encKey)=>{
+    return !encKey.deleted;
+  });
+  if(!encKey){
+    throw new Error('no encKey found');
+  }
+
+  var encKeyId = encKey.id;
+  var data = tokenCrypt.encrypt(str, encKey.key);
+  return { encKeyId, data };
+};
+
+const decryptStrUsingOrgEncKey = ({ data, encKeyId, org })=>{
+  if(!data || !encKeyId || !org){
+    throw new Error('needs { data, encKeyId, org } properties');
+  }
+  var encKey = _.find(org.encKeys||[], (e)=>{
+    return (e.id == encKeyId);
+  });
+  if(!encKey){
+    throw new Error('no matching encKey found');
+  }
+  return tokenCrypt.decrypt(data, encKey.key);
+};
+
+var genKey = ()=>{
+  var bytes = 32;
+  var randBuff = crypto.randomBytes(bytes);
+  if(!randBuff[0]){
+    randBuff[0] = _.random(1, 255);
+  }
+  if(!randBuff[bytes - 1]){
+    randBuff[bytes - 1] = _.random(1, 255);
+  }
+  var key = randBuff.toString('base64');
+  var id = uuid();
+  var creationTime = new Date();
+  var deleted = false;
+  return {
+    id, key, creationTime, deleted,
+  };
+};
+
+var cronRotateEncKeys = async({ db, maxAge=1000*60*60*24*365/2 })=>{
```

Comment:
How do we ensure that we don't rotate the keys while we are already in the process of rotating the keys